outputs.tf

```diff
@@ -93,11 +93,26 @@ output "cluster_ca_certificate" {
   value       = azurerm_kubernetes_cluster.main.kube_config[0].cluster_ca_certificate
 }
 
+output "cluster_fqdn" {
+  description = "The FQDN of the Azure Kubernetes Managed Cluster."
+  value       = azurerm_kubernetes_cluster.main.fqdn
+}
+
 output "cluster_identity" {
   description = "The `azurerm_kubernetes_cluster`'s `identity` block."
   value       = try(azurerm_kubernetes_cluster.main.identity[0], null)
 }
 
+output "cluster_portal_fqdn" {
+  description = "The FQDN for the Azure Portal resources when private link has been enabled, which is only resolvable inside the Virtual Network used by the Kubernetes Cluster."
+  value       = azurerm_kubernetes_cluster.main.portal_fqdn
+}
+
+output "cluster_private_fqdn" {
+  description = "The FQDN for the Kubernetes Cluster when private link has been enabled, which is only resolvable inside the Virtual Network used by the Kubernetes Cluster."
+  value       = azurerm_kubernetes_cluster.main.private_fqdn
```

Comment:
Hi @zioproto sorry for the late reply, I've tested and confirmed that this `private_fqdn` will be empty string when we set aks's `private_cluster_enabled` to `false`.

Reply:
OK. Do you think is necessary to add a comment to the function `assertOutputNotEmpty` to clarify that this function can be used only when testing with private clusters ?
Let me know if it makes sense to add this comment in the code.
If in the future there will be a scenario with public clusters we want to make sure that `assertOutputNotEmpty` is not used because it will always fail. Thank you !

Reply:
I think it's ok since output's description has described this output only works for private cluster, if we change this example to provision a public cluster in the future and somehow we forget to remove this `assertOutputNotEmpty` from our e2e test, the failed test will warn us and the we can figure out what's wrong quickly by reading description.